File: backend/app/schemas/stats.py
from pydantic import BaseModel
from typing import Optional


class UserStatsResponse(BaseModel):
    username: str
    total_games: int
    wins: int
    losses: int
    in_progress: int
    win_rate: float


class LeaderboardEntry(BaseModel):
    rank: int
    username: str
    wins: int
    total_games: int
    win_rate: float


class LeaderboardMeta(BaseModel):
    page: int
    page_size: int
    total_entries: int
    total_pages: int
    tier: Optional[int]
    tier_label: str
    available_tiers: list[int]
    total_users: int


class PaginatedLeaderboardResponse(BaseModel):
    entries: list[LeaderboardEntry]
    meta: LeaderboardMeta


class UserRankResponse(BaseModel):
    rank: int
    username: str
    wins: int
    total_games: int
    win_rate: float
    total_users: int
    percentile: float


# Keep for backward compat if anything imports this
class LeaderboardResponse(BaseModel):
    entries: list[LeaderboardEntry]

File: frontend/src/types/stats.ts
export interface UserStats {
  username: string
  total_games: number
  wins: number
  losses: number
  in_progress: number
  win_rate: number
}

export interface LeaderboardEntry {
  rank: number
  username: string
  wins: number
  total_games: number
  win_rate: number
}

export interface LeaderboardMeta {
  page: number
  page_size: number
  total_entries: number
  total_pages: number
  tier: number | null
  tier_label: string
  available_tiers: number[]
  total_users: number
}

export interface PaginatedLeaderboardResponse {
  entries: LeaderboardEntry[]
  meta: LeaderboardMeta
}

export interface UserRankResponse {
  rank: number
  username: string
  wins: number
  total_games: number
  win_rate: number
  total_users: number
  percentile: number
}